Source/ACE.Server/WorldObjects/WorldObject_Use.cs

```diff
@@ -92,12 +93,24 @@ public virtual void OnActivate(WorldObject activator)
                 return;
             }
 
+            if (player != null)
+                player.EnchantmentManager.StartCooldown(this);
+
             // if ActivationTarget is another object,
             // should this be checking the ActivationResponse of the target object?
 
-            // default use action
-            if (ActivationResponse.HasFlag(ActivationResponse.Use))
+            // Chests with 'ActivationResponse - CastSpell' should still ActOnUse
```

Comment:
this is the type of logic that shouldn't be in base

this is what the overrides are for, subtype-specific logic

Reply:
Except when the code was as it was before, the chests didn't actually do anything.  Merely allowing them past the Active check, the chest would cast its spell every time the interacted with the chest in any way, as the flow would perform the ActOnUse and then the OnCastSpell would fire.  Should only be casting the spell when opening the chest.

Reply:
the fixes for these comments have been posted as a patch in the dev channel

Reply:
Your suggestions for OnActivate() won't work for Chests, as the spell is only cast when a player opens the chest, not for all interactions.